tests/test_unit/test_auth/test_auth_keycloak.py

```diff
@@ -0,0 +1,151 @@
+import pytest
+import responses
+from httpx import AsyncClient
+
+from syncmaster.backend.settings import BackendSettings as Settings
+from tests.mocks import MockUser
+
+KEYCLOAK_PROVIDER = "syncmaster.backend.providers.auth.keycloak_provider.KeycloakAuthProvider"
+pytestmark = [pytest.mark.asyncio, pytest.mark.backend]
+
+
+@responses.activate
+@pytest.mark.parametrize(
+    "settings",
+    [
+        {
+            "auth": {
+                "provider": KEYCLOAK_PROVIDER,
+                "server_url": "http://localhost:8080/",
+            },
+        },
+    ],
+    indirect=True,
+)
+async def test_get_keycloak_user_unauthorized(client: AsyncClient, mock_keycloak_well_known):
+    response = await client.get("/v1/users/some_user_id")
+
+    # redirect unauthorized user to Keycloak
+    assert response.status_code == 307
+    assert "protocol/openid-connect/auth?" in str(
+        response.next_request.url,
+    )
+
+
+@responses.activate
+@pytest.mark.parametrize(
+    "settings",
+    [
+        {
+            "auth": {
+                "provider": KEYCLOAK_PROVIDER,
+                "server_url": "http://localhost:8080/",
+            },
+        },
+    ],
+    indirect=True,
+)
+async def test_get_keycloak_user_authorized(
+    client: AsyncClient,
+    simple_user: MockUser,
+    settings: Settings,
+    create_session_cookie,
+    mock_keycloak_well_known,
+    mock_keycloak_realm,
+):
+    session_cookie = create_session_cookie(simple_user)
+    headers = {
+        "Cookie": f"session={session_cookie}",
+    }
+    response = await client.get(
+        f"/v1/users/{simple_user.id}",
+        headers=headers,
+    )
+
+    assert response.status_code == 200
+    assert response.json() == {
+        "id": simple_user.id,
+        "is_superuser": simple_user.is_superuser,
+        "username": simple_user.username,
+    }
+
+
+@responses.activate
+@pytest.mark.parametrize(
+    "settings",
+    [
+        {
+            "auth": {
+                "provider": KEYCLOAK_PROVIDER,
+                "server_url": "http://localhost:8080/",
+            },
+        },
+    ],
+    indirect=True,
+)
+async def test_get_keycloak_deleted_user(
+    client: AsyncClient,
+    simple_user: MockUser,
+    deleted_user: MockUser,
+    settings: Settings,
+    create_session_cookie,
+    mock_keycloak_well_known,
+    mock_keycloak_realm,
+):
+    session_cookie = create_session_cookie(simple_user)
+    headers = {
+        "Cookie": f"session={session_cookie}",
+    }
+    response = await client.get(
+        f"/v1/users/{deleted_user.id}",
```

Comment:
What if cookie is set and not expired, but user is disabled/blocked in Keycloak?

Reply:
then in worst case he will have access to resource until his access_token is not expired (in mts infrastructure it is not more than 15 minutes)

Reply:
I think we should add a comment about that somethere